### PR TITLE
Updating CI Process Instructions + Updating Deployment Instructions

### PR DESCRIPTION
**Pull request by sunithasomanna**
Updating readme.md and replacing the existing "CI Process" section with new content and screenshots provided by Rahul.

### CODE REVIEW

**Review by rahulroshan-kachchap — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,798 +4,3963 @@
...
+        It has only one controller node that also functions as a worker node. It does not support scaling of the resources and services of Resource Aggregator for ODIM—you cannot add nodes into a one-node cluster.
 
-The following figure shows these functional layers of Resource Aggregator for ODIM deployed in a data center.
+    -  Three-node cluster:
+	
+        It has three controller nodes that also function as worker nodes and additional worker nodes for sharing extra load. It provides High Availability \(HA\) by allowing scaling of the resources and services of Resource Aggregator for ODIM—you can add worker nodes and increase the number of service instances running in a cluster.
 
-![ODIM_architecture](images/arch.png)
+        <blockquote>
+        NOTE: The total number of nodes in a cluster must be odd.
```
Comment (bharath-b-hpe) [inline README.md:196]: Note is not required, as we support only one or three node deployment.

File: README.md [hunk 2]
```diff
@@ -4,798 +4,3963 @@
...
+   |quay.io/coreos/etcd|v3.4.3 |quay.io_coreos_etcd.tar |
+   
+2. Verify that the images are successfully pulled using the following command:
    ```
-   $ ps -eaf | grep svc
+   $ docker images
    ```
+3. Save each Docker image to a tar archive using the following command:
+    ```
+    $ sudo docker save -o <image_name.tar> <image_name:tag>
```
Comment (bharath-b-hpe) [inline README.md:409]: A note to refer the table above for the mapping of tar archive name for each image should be helpful

File: README.md [hunk 3]
```diff
@@ -4,798 +4,3963 @@
...
+	   $ export ODIMRA_GROUP_ID=2021
+     ```
+   5. ```
+      $ ./build_images.sh
+      ```
+   6. ```
+      $ sudo docker images
+	   ```
+	    If the images are built successfully, you will receive an output which is similar to the following sample:
+	  ```
```
Comment (bharath-b-hpe) [inline README.md:469]: output is enclosed in code block, please check

File: README.md [hunk 4]
```diff
@@ -4,798 +4,3963 @@
...
+      events                                    1.0                                     860a9202a483              22 hours ago                  130MB
+      api                                       1.0                                     effab530ede5                22 hours ago                  130MB
+      aggregation                               1.0                                     354f67a857b6               22 hours ago                  130MB
+	   account-session                           1.0                                     a7eb07e69395              22 hours ago                  129MB
+	  ```
+2. Save each Docker image to a tar archive using the following command:
+    ```
+    $ sudo docker save -o <image_name.tar> <image_name:tag>
+    ```
+	Example: `sudo docker save -o consul.tar consul:1.0`
```
Comment (bharath-b-hpe) [inline README.md:490]: Need to add a table with the list of ODIMRA images and corresponding archive name to be used for saving the images.

**Comment by sunithasomanna** [comment]
Hi Bharath, please review the PR. Addressed your previous comments.

**Review by bharath-b-hpe — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1,801 +1,3854 @@
...
+
+   1. Check if the Docker group exists using the following command:
+      ```
+	    $ getent group docker
+	   ```
+	  ```
+	
+     ```
+   
+	  Example output: `docker:x:998:<username>`
```
Comment (sukhwinder20) [inline README.md:2164]: output is not showing username:
bruce@dnode1:~$ getent group docker
docker:x:999:
bruce@dnode1:~$

File: README.md [hunk 6]
```diff
@@ -1,801 +1,3854 @@
...
+1. Download and install Ubuntu 18.04 LTS on the deployment node and all the cluster nodes. 
+    During installation, configure the IP addresses of cluster nodes to reach the management VLANs where devices are connected. Ensure there is no firewall or switches blocking the connections and ports.
+
+   <blockquote>
+    IMPORTANT: Ensure you create the same non-root username and password on all the cluster nodes during the installation of OS.
+    </blockquote>
+2. Verify that the time across all the nodes are synchronized. To know how to set up time sync, see [Setting up time sync across nodes](#setting-up-time-sync-across-nodes). 
+3. Run the following commands to install packages such as Python, Java, Ansible, and others on the deployment node: 
+
+   Before running the following commands, ensure you are able to download the external packages through `apt-get`. If the nodes are behind a corporate proxy or firewall, set your proxy configuration. To know how to set proxy, see [Setting proxy configuration](#setting-proxy-configuration).
```
Comment (sukhwinder20) [inline README.md:257]: Proxy should be enabled on deployer and cluster node

File: README.md [hunk 7]
```diff
@@ -1,801 +1,3854 @@
...
+      $ cd ODIM
       ```
-    b. Scroll to the end of the file, add the following line, and then save:
+      
+	3. ```
+      $ export ODIMRA_USER_ID=2021
       ```
-      <host_ipv4_address> <user_preferred_fqdn_for_host>
+	   
+   4. ```
```
Comment (sukhwinder20) [inline README.md:449]: Remove these commas

File: README.md [hunk 8]
```diff
@@ -1,801 +1,3854 @@
...
+	   | urplugin        | 1.0     | fb3c1cf141d5 | 6 weeks ago  | 101MB    |
+	   | api             | 1.0     | effab530ede5 | 2 months ago | 130MB    |
+	   | aggregation     | 1.0     | 354f67a857b6 | 3 months ago | 130MB    |
+	   | account-session | 1.0     | a7eb07e69395 | 3 months ago | 129MB    |
+	
+2. Save each Docker image to a tar archive using the following command:
+    ```
+    $ sudo docker save -o <image_name.tar> <image_name:tag>
+    ```
+	Example: `sudo docker save -o consul.tar consul:1.0`
```
Comment (sukhwinder20) [inline README.md:486]: consul tag is 1.6 so change example with sudo docker save -o consul.tar consul:1.6

File: README.md [hunk 9]
```diff
@@ -1,801 +1,3854 @@
...
+	| fabrics               | 1.0         | fabrics.tar                  |
+	| managers              | 1.0         | managers.tar                 |
+	| systems               | 1.0         | systems.tar                  |
+	| task                  | 1.0         | task.tar                     |
+	| update                | 1.0         | update.tar                   |
+	| kafka                 | 1.0         | kafka.tar                    |
+	| zookeeper             | 1.0         | zookeeper.tar                |
+	| consul                | 1.6         | consul.tar                   |
+	| redis                 | 1.0         | redis.tar                    |
+	| stakater/reloader     | v0.0.76     | stakater_reloader.tar        |
```
Comment (sukhwinder20) [inline README.md:505]: stakater/reloader is not available in image list

File: README.md [hunk 10]
```diff
@@ -1,801 +1,3854 @@
...
+	| busybox               | 1.33        | busybox.tar                  |
```
Comment (sukhwinder20) [inline README.md:506]: busybox is not available in image list

File: README.md [hunk 11]
```diff
@@ -1,801 +1,3854 @@
...
+	   | events          | 1.0     | 860a9202a483 | 11 days ago  | 130MB    |
+	   | aciplugin       | 1.0     | 0c42ba5d4223 | 6 weeks ago  | 114MB    |
+	   | urplugin        | 1.0     | fb3c1cf141d5 | 6 weeks ago  | 101MB    |
+	   | api             | 1.0     | effab530ede5 | 2 months ago | 130MB    |
+	   | aggregation     | 1.0     | 354f67a857b6 | 3 months ago | 130MB    |
+	   | account-session | 1.0     | a7eb07e69395 | 3 months ago | 129MB    |
+	
+2. Save each Docker image to a tar archive using the following command:
+    ```
+    $ sudo docker save -o <image_name.tar> <image_name:tag>
```
Comment (sukhwinder20) [inline README.md:484]: odim.tar is not required here

File: README.md [hunk 12]
```diff
@@ -1,801 +1,3854 @@
...
+      ```
+      $ cd ~/ODIM/odim-controller/scripts
+      ```
 
-4.   Restart Docker using the following command: 
+   2. Open the `kube_deploy_nodes.yaml` file to edit: 
 
-     ```
-     $ docker restart build_odimra_1
       ```
```
Comment (sukhwinder20) [inline README.md:633]: cp kube_deploy_nodes.yaml.tmpl  kube_deploy_nodes.yaml

File: README.md [hunk 13]
```diff
@@ -1,801 +1,3854 @@
...
+    |<plugin_name>|Name of the plugin. Example: "grfplugin"<br>|
+    |<k8s_self_node_IP>|The physical IP address of the cluster node.|
+    |<k8s_node2_IP><k8s_node3_IP><br>|The physical IP addresses of the other cluster nodes.|
+    |<plugin_node_port>|The port specified for the eventListenerNodePort configuration parameter in the `<plugin_name>-config.yaml` file.|
+    |<VIP>|Virtual IP address specified in the keepalived.conf file.|
+    |<nginx_plugin_port>|Any free port on the cluster node. It must be available on all the other cluster nodes.Preferred port is above 45000.<br>Ensure that this port is not used as any other service port.<br><blockquote>NOTE: You can reach the resource aggregator API server at:<br>`https://<VIP>:<nginx_api_port>`.<br></blockquote>|
+
+4. Restart Nginx systemd service only on the leader node \(cluster node where Keepalived priority is set to a higher number\): 
+
+    ```
```
Comment (sukhwinder20) [inline README.md:2977]: sudo systemctl restart keepalived

File: README.md [hunk 14]
```diff
@@ -1,801 +1,3854 @@
...
+    upstream <plugin_name>  {
+      server <k8s_self_node_IP>:<plugin_node_port> max_fails=2 fail_timeout=10s;
+      server <k8s_node2_IP>:<plugin_node_port> max_fails=2 fail_timeout=10s backup;
+      server <k8s_node3_IP>:<plugin_node_port> max_fails=2 fail_timeout=10s backup;
+    }
+     
+    server {
+            listen <k8s_self_node_IP>:<nginx_plugin_port> ssl;
+            listen <VIP>:<nginx_plugin_port>** ssl;
```
Comment (sukhwinder20) [inline README.md:2942]: ** needs to be removed

File: README.md [hunk 15]
```diff
@@ -1,801 +1,3854 @@
...
+2. Log in to each cluster node and run the following commands: 
+
+    ```
+    $ sudo mkdir -p /var/log/grfplugin_logs/
+    ```
+
+    ```
+    $ sudo chown odimra:odimra /var/log/grfplugin_logs/
+    ```
+
```
Comment (sukhwinder20) [inline README.md:3284]: Login to deployer node

File: README.md [hunk 16]
```diff
@@ -1,801 +1,3854 @@
...
+4. To scale up the resource aggregator deployments, run the following command on the deployment node: 
+
+    ```
+    $ python3 odim-controller.py --config \
+     /home/${USER}/ODIM/odim-controller/scripts/kube_deploy_nodes.yaml \
+     --scale --svc <deployment_name> --replicas <replica_count>
+    ```
+
+    Replace <deployment\_name\> with the name of the deployment which you want to scale up. To know all the supported deployment names, see [Resource Aggregator for ODIM deployment names](#resource-aggregator-for-odim-deployment-names).
```
Comment (sukhwinder20) [inline README.md:1767]: Deployment name is not the pre-defined name here. Its  given by user

File: README.md [hunk 17]
```diff
@@ -1,801 +1,3854 @@
...
+
+1. ```
+    $ python3 odim-controller.py --addnode kubernetes --config \
+    ~/ODIM/odim-controller/scripts/kube_deploy_nodes.yaml
+   ```
+   
+2. ```
+     $ python3 odim-controller.py --config \
+    ~/ODIM/odim-controller/scripts/kube_deploy_nodes.yaml \
+    --scale --svc threenodecluster --replicas 3
```
Comment (sukhwinder20) [inline README.md:1717]: threenodecluster is not there in deployment list

**Review by sukhwinder20 — APPROVED** [review]

**Pull request merged**